### ISSUE TITLE
Polycentric comments aren't posted

### ISSUE THREAD

**Issue by mathix420** [futo-org/grayjay-android#1074]
Posting a comment that appears in my comment history but not in the video comment section nor my polycentric profile.

Let me know if you need logs or additional details.

**Comment by mathix420** [comment]
Maybe this is linked ![Screenshot_20240615-003040.png](https://github.com/futo-org/grayjay-android/assets/37625778/ddbf5c1e-80b0-4595-8171-fdaf839eb109)

**Comment by EzraM5** [comment]
I've had exactly the same issues. Here's the kinds of things I've noticed when looking into this myself:

1. This issue pops up if you are copying an ID from one version of the app to another. I noticed this between trying both the Website's download and the F-Droid version of the app.
2. The way I believe I've gotten around this was going to Polycentric.io and actually making an identity there before logging into Grayjay with it.. I have yet to receive that error since doing so, but I am not 100% sure.

I don't know if that will help you or anyone, but it's helped me so far.

**Comment by mathix420** [comment]
Yes I've done that as well, and it has fixed the error message but not the comment issues sadly

**Comment by Generalaccountuse** [comment]
What gives, this is happening to me too. My comments aren't being posted on the videos i watch.

**Comment by golgothan666** [comment]
just noticed this is happening to me as well... starting about 3 weeks ago, also

and, i'm unable to export the Polycentric key... just freezes/crashes

**Comment by TheWheee** [comment]
im not sure why but now the comments appeared

**Comment by ericcox** [comment]
Keep in mind, youtube is censoring heavily right now, due to the election.  And the scenario above is exactly what happens when a comment is censored: after you make a comment (not through Grayjay - directly on YT), if you reload the video, the comment is gone. It appears in your history but not in the video comment section.  Sometimes it appears later.  I comment directly through the YT site and this happens to me all the time.

**Comment by Bluscream** [comment]
@ericcox Keep in mind, polycentric comments have nothing to do with Youtube and Youtube has no control over them. They can only censor platform comments, which you can't post with GrayJay anyway. The timeout to the *.polycentric.io domain can only mean two things, either it's actually a network timeout (your device can't send the comment packet to the polycentric server in time for some reason) or the packet reaches the polycentric server but it takes too long to process it and gives up

**Comment by johnsmith744** [comment]
+1 on my comment not showing.  Come on Grayjay, this is pathetic.

**Comment by treyus30** [comment]
Does Greyjay cache the comments it thinks you sent to Polycentric? I can't think of any other reason why they would show in your history.

BTW, I confirmed that if you create a new identity, your first comment and that comment alone will post. You can't delete, you can't comment on your comment, you can't post more. But they will all show in history.

**Comment by Bluscream** [comment]
> This is the same as shadow banning and removing comment, it IS what youtube is doing.

@johnsmith744 No. It might have the same result, but it's not the same thing as censorship if it's unintentional.

**Comment by Pyrate37** [comment]
Same issue.

**Comment by Blackilykat** [comment]
This is happening for me as well, but I didn't see any error. It started about a week ago. The latest comment with likes shows up when I click it to open the video it was posted on, while all the ones with no interactions after do not appear.

I'm unsure about this but it may have started when I exported my polycentric identity (so I could log in on the website)

I had this account for months and never got many dislikes (most dislikes were 2, worst like/dislike ratio was 0:1)

**Comment by TheQuickFox** [comment]
https://github.com/futo-org/grayjay-android/issues/1202

How do I get support on this problem? It does not look like any of the devs are commenting here.
Is there a GrayJay support desk or support e-mail?

**Comment by canadiannomad** [comment]
Having this too..
I can see my comments in the history in the app, but not in the history on the platform, so presumably it is because it could not send it, for whatever reason, to the server.
If that is true, then I think the real issue is that there needs to be the ability to resend comments when they fail.  Possibly automatically, but even manually would be nice.

**Comment by johnsmith744** [comment]
> Having this too.. I can see my comments in the history in the app, but not in the history on the platform, so presumably it is because it could not send it, for whatever reason, to the server. If that is true, then I think the real issue is that there needs to be the ability to resend comments when they fail. Possibly automatically, but even manually would be nice.

you can repost it as many time as you want, it still doesn't get post.

**Comment by canadiannomad** [comment]
> you can repost it as many time as you want, it still doesn't get post.

Is there a retry button I don't know about?

Agreed, once grayjay decided not to post my comments, no new comments would work, even after logout/login.  But posting the same comments via the polycentric web app works.

**Comment by Generalaccountuse** [comment]
I am just giving up on GrayJay.
i might as well just go back to Newpipe if I cant make comments on GrayJay, since I can't make more Poly IDs on my phone.

Apparently there is a limit.

Sent with [Proton Mail](https://proton.me/) secure email.

**Comment by mathix420** [comment]
It happened again for me, I think I have more insight to possibly find the source of the issue.

1. Post a comment on a video
2. Delete this comment
3. Post again on the same video

And suddenly none of your comments are being published...

**Comment by Bluscream** [comment]
You will most likely not reach them here, use the zulip chat instead

**Comment by Blackilykat** [comment]
> Also poly comments are only seen by 0.0001% of viewers and not even by the creator so your comments will mostly be seen by yourself only.
Most videos I watch have at least 1-2 likes on grayjay, with some reaching hundreds. Even if it's just a few people compared to the people on youtube, if I make a comment on grayjay I want it to show up for those few people.
But anyway, I don't think discussing the popularity of grayjay as a platform is relevant to this issue.

**Comment by treyus30** [comment]
>  Also they stated somewhere that they are now emploing AI to filter illegal content/hate speech so you might have just been banned/filtered but no way to say for sure. 

What would the point of this be? To become exactly like the platform they're allowing us to escape?

**Comment by EzraM5** [comment]
> You will most likely not reach them here, use the zulip chat instead. Also they stated somewhere that they are now emploing AI to filter illegal content/hate speech so you might have just been banned/filtered but no way to say for sure. Also poly comments are only seen by 0.0001% of viewers and not even by the creator so your comments will mostly be seen by yourself only. If there was a browser extension that would load poly comments alongside regular youtube it would maybe be a bit of a different story but there just isnt

Lol

Lmao

AI to moderate. What a goddamn joke. It's not like YouTube didn't already try to do that. It's even more hilarious given how easy it is to just up and make a new account.

Welp. Thank you for that. Good to know that this whole thing ain't worth bothering with. Saves me a lot of time wasted.

**Comment by Bluscream** [comment]
> It's not like this app will ever be accepted by any mainstream app store anyway..

https://play.google.com/store/apps/details?id=com.futo.platformplayer.playstore

**Comment by EzraM5** [comment]
> > It's not like this app will ever be accepted by any mainstream app store anyway..
> 
> https://play.google.com/store/apps/details?id=com.futo.platformplayer.playstore

It won't, but like

I still think of Pirate Software's opinion about it:

If you want people's general opinion to be good, you really gotta take care of it yourself. Otherwise, people are just going to avoid it like the plague.

And having been on Polycentric for a little while, it absolutely is filled with just the worst people imaginable.

**Comment by Bluscream** [comment]
> And having been on Polycentric for a little while, it absolutely is filled with just the worst people imaginable.

That's just what happens on any platform that is competing with youtube, and sometimes on youtube themselves, for me the people commenting is reason why im never clicking on the poly comments tab and always look at platform comments.

**Comment by EzraM5** [comment]
> > And having been on Polycentric for a little while, it absolutely is filled with just the worst people imaginable.
> 
> That's just what happens on any platform that is competing with youtube, and sometimes on youtube themselves, for me the people commenting is reason why im never clicking on the poly comments tab and always look at platform comments.

Fair enough.

I'm just saying that it REALLY doesn't have to be that way. There are tools to deal with that kind of garbage as a whole.

**Comment by Generalaccountuse** [comment]
Everyone wants to say there is speech garbage, but at the same time want to fall on the sword for free speech; hypocrites.
With the exception of blatant insults and descriptive pornography and violence:
Speech should be allowed.

What most people call "hate speech" is factual pattern based descriptions of behaviors from certain demographics.
The Truth hurts, so buy some earplugs if you want to live a sheltered life.

The fact that i knew my comments and account was bring targetted and banned, by GrayJay, since some user was stalking me and because I decided to "verbally" accost him for it DAILY; shows me the "hate speech" users are correct in denouncing GrayJay.

I thought this new app really had potential for the future to bring us out of totalitarian power abuse social media platforms have unstoppably been ushering into the 21st century, people would just have needed to put in the time and effort for it; but it seems I was wrong.
Bad things are in store for the future because of this over sensitive generation and they will share the proper blame for its horrible painful events just as much if not more than the "hate speech" members you so like to vilify.

Oh well, History repeats itself, and every human is to blame for it.

Good bye fake freedom of speech grajay, and good bye to it's sensitive members.

**Comment by EzraM5** [comment]
> Everyone wants to say there is speech garbage, but at the same time want to fall on the sword for free speech; hypocrites. With the exception of blatant insults and descriptive pornography and violence: Speech should be allowed.
> 
> What most people call "hate speech" is factual pattern based descriptions of behaviors from certain demographics. The Truth hurts, so buy some earplugs if you want to live a sheltered life.
> 
> The fact that i knew my comments and account was bring targetted and banned, by GrayJay, since some user was stalking me and because I decided to "verbally" accost him for it DAILY; shows me the "hate speech" users are correct in denouncing GrayJay.
> 
> I thought this new app really had potential for the future to bring us out of totalitarian power abuse social media platforms have unstoppably been ushering into the 21st century, people would just have needed to put in the time and effort for it; but it seems I was wrong. Bad things are in store for the future because of this over sensitive generation and they will share the proper blame for its horrible painful events just as much if not more than the "hate speech" members you so like to vilify.
> 
> Oh well, History repeats itself, and every human is to blame for it.
> 
> Good bye fake freedom of speech grajay, and good bye to it's sensitive members.

Christ, dude, that's a whole lot of words to say "nobody likes it when I say the N word. :("

Have you tried not being racist? It's pretty easy. You can just NOT hate brown people.

**Comment by Generalaccountuse** [comment]
Christ, dude, that's a whole lot of words to say "nobody likes it when I say the N word. :("
} I've never said, and don't say that word. If reading words hurts you that much, try staying away from intelligent conversations; trust me, it'll hurt more.

Have you tried not being racist?
} have you tried thinking?
Tell me your made up definition of that word, and we'll see what I am and what you are as well.

 It's pretty easy. 
} yea, so is critical thinking; but you can't even manage that, so you're sol.

You can just NOT hate brown people.
} you can just NOT hate white people.
Have you tried that?

See what I did there??
Interesting how it goes when you get strawmanned back, isn't it??

**Comment by EzraM5** [comment]
> Christ, dude, that's a whole lot of words to say "nobody likes it when I say the N word. :(" } I've never said, and don't say that word. If reading words hurts you that much, try staying away from intelligent conversations; trust me, it'll hurt more.
> 
> Have you tried not being racist? } have you tried thinking? Tell me your made up definition of that word, and we'll see what I am and what you are as well.
> 
> It's pretty easy. } yea, so is critical thinking; but you can't even manage that, so you're sol.
> 
> You can just NOT hate brown people. } you can just NOT hate white people. Have you tried that?
> 
> See what I did there?? Interesting how it goes when you get strawmanned back, isn't it??

1.: Don't believe you, because two:

2.: "Have you tried thinking?: People who usually say this think that this is some excuse from criticism against their own flawed thinking. It's not. It's not hard to look at someone being racist and think "Man, maybe I don't want to be a complete piece of garbage!"

3. Critical thinking: Don't know what that has to do with anything. See 2.

4. Not hating brown people: I am brown people. Pretty sure I don't hate myself, and we make pretty racy jokes about ourselves without your help, thanks.

**Comment by Generalaccountuse** [comment]
1.: Don't believe you, because two:
} that's your problem to fix, not mine.

2.: "Have you tried thinking?: People who usually say this 
} its good to see that you haven't stopped making assumptions about me, and people in general, ever since you started replying; typical mentality of a non critical thinker and someone who is a closet racist.
I've met plenty like you, but trust me, you've never met any like me.

think that this is some excuse from criticism against their own flawed thinking.
} kinda like yours.
:)

 It's not. It's not hard to look at someone being racist 
} who hurt you??
That you are on such a crusade against mUh RaCIsT people; yet you have had the mental capacity or even fortitude to answer my request to give me your definition of it; * LAUGIING IN MY MIND*!

and think "Man, maybe I don't want to be a complete piece of garbage!"
} You sure are kid.

Critical thinking: Don't know what that has to do with anything. See 2.
} yes, clearly you don't because you can not; sucks to be you.


Not hating brown people: I am brown people.
} I never asked what you were, but apart from not being able to think, you also can't read, I'm not surprised.

 Pretty sure I don't hate myself, and we make pretty racy jokes
} That's what all your "RACIST" PTSD is from, racy jokes?!?!
Lol, awwwww man, I knew I was talking to a girl!!?

 about ourselves without your help, thanks.
} your welcome.
:)

**Comment by EzraM5** [comment]
> 1.: Don't believe you, because two: } that's your problem to fix, not mine.
> 
> 2.: "Have you tried thinking?: People who usually say this } its good to see that you haven't stopped making assumptions about me, and people in general, ever since you started replying; typical mentality of a non critical thinker and someone who is a closet racist. I've met plenty like you, but trust me, you've never met any like me.
> 
> think that this is some excuse from criticism against their own flawed thinking. } kinda like yours. :)
> 
> It's not. It's not hard to look at someone being racist } who hurt you?? That you are on such a crusade against mUh RaCIsT people; yet you have not had the mental capacity or even the fortitude to answer my request to give me your definition of it; * LAUGIING IN MY MIND*!
> 
> and think "Man, maybe I don't want to be a complete piece of garbage!" } You sure are kid.
> 
> Critical thinking: Don't know what that has to do with anything. See 2. } yes, clearly you don't because you can not; sucks to be you.
> 
> Not hating brown people: I am brown people. } I never asked what you were, but apart from not being able to think, you also can't read, I'm not surprised.
> 
> Pretty sure I don't hate myself, and we make pretty racy jokes } That's what all your "RACIST" PTSD is from, racy jokes?!?! Lol, awwwww man, I knew I was talking to a girl!!?
> 
> about ourselves without your help, thanks. } your welcome. :)

TL;DC: Only proving me right by wasting your time writing all that nonsense.

**Comment by treyus30** [comment]
> > Everyone wants to say there is speech garbage, but at the same time want to fall on the sword for free speech; hypocrites. With the exception of blatant insults and descriptive pornography and violence: Speech should be allowed.
> > 
> > What most people call "hate speech" is factual pattern based descriptions of behaviors from certain demographics. The Truth hurts, so buy some earplugs if you want to live a sheltered life.
> > 
> > The fact that i knew my comments and account was bring targetted and banned, by GrayJay, since some user was stalking me and because I decided to "verbally" accost him for it DAILY; shows me the "hate speech" users are correct in denouncing GrayJay.
> > 
> > I thought this new app really had potential for the future to bring us out of totalitarian power abuse social media platforms have unstoppably been ushering into the 21st century, people would just have needed to put in the time and effort for it; but it seems I was wrong. Bad things are in store for the future because of this over sensitive generation and they will share the proper blame for its horrible painful events just as much if not more than the "hate speech" members you so like to vilify.
> > 
> > Oh well, History repeats itself, and every human is to blame for it.
> > 
> > Good bye fake freedom of speech grajay, and good bye to it's sensitive members.
> 
> Christ, dude, that's a whole lot of words to say "nobody likes it when I say the N word. :("
> 
> Have you tried not being racist? It's pretty easy. You can just NOT hate brown people.

Use your poorly formulated bait on Reddit or Facebook next time, not here. Certain words don't mean you hate someone or groups of someones.

**Comment by Blackilykat** [comment]
guys can you take your argument somewhere else? you've been filling my inbox with stuff unrelated to this issue.

**Comment by mathix420** [comment]
I finally find a fix tor this issue!

1. Settings > Scroll down > Backup > Export data manually
2. Manage polycentric identity > Export > Copy backup string and store it in password manager
3. Wipe all the app data in your phone's settings
4. Open GrayJay
5. Enable the Sources you were using before wiping data (or else they will all be enabled)
6. Settings > Import Data > Choose the zip you have exported in step 1
7. Follow instruction, Click No when asked to enable all sources (I find it annoying to have them all enabled).
8. Settings > Manage polycentric id > Import your ID back into GrayJay by pasting the backup string
9. YOU CAN FINALLY USE THIS APP AS INTENDED BACK AGAIN!

**Comment by johnsmith744** [comment]
> Everyone wants to say there is speech garbage, but at the same time want to fall on the sword for free speech; hypocrites. With the exception of blatant insults and descriptive pornography and violence: Speech should be allowed.
> 
> What most people call "hate speech" is factual pattern based descriptions of behaviors from certain demographics. The Truth hurts, so buy some earplugs if you want to live a sheltered life.
> 
> The fact that i knew my comments and account was bring targetted and banned, by GrayJay, since some user was stalking me and because I decided to "verbally" accost him for it DAILY; shows me the "hate speech" users are correct in denouncing GrayJay.
> 
> I thought this new app really had potential for the future to bring us out of totalitarian power abuse social media platforms have unstoppably been ushering into the 21st century, people would just have needed to put in the time and effort for it; but it seems I was wrong. Bad things are in store for the future because of this over sensitive generation and they will share the proper blame for its horrible painful events just as much if not more than the "hate speech" members you so like to vilify.
> 
> Oh well, History repeats itself, and every human is to blame for it.
> 
> Good bye fake freedom of speech grajay, and good bye to it's sensitive members.

I had an "AI" stalking me too.  I was ignoring him and advising people not to bother with him when they try to defend me.  I would  put him on "block" if greyjay have that feature.  "Block" is an individual choice, unlike "ban" or "shadow ban" is by the "overlord".

I have no evidence that greyjay is doing this on purpose but right now I don't see a better choice.  Just here looking for a solution to a problem I have.

**Comment by TheQuickFox** [comment]
> I finally found a fix tor this issue!
> 
>     1. Settings > Scroll down > Backup > Export data manually
> 
>     2. Manage polycentric identity > Export > Copy backup string and store it in password manager
> 
>     3. Wipe all the app data in your phone's settings
> 
>     4. Open GrayJay
> 
>     5. Enable the Sources you were using before wiping data (or else they will all be enabled)
> 
>     6. Settings > Import Data > Choose the zip you have exported in step 1
> 
>     7. Follow instruction, Click No when asked to enable all sources (I find it annoying to have them all enabled).
> 
>     8. Settings > Manage polycentric id > Import your ID back into GrayJay by pasting the backup string
> 
>     9. YOU CAN FINALLY USE THIS APP AS INTENDED BACK AGAIN!
> 
> 
> My guess is that the issue happen when you're deleting some comments, so don't until futo take care of this issue..
> 
> ~Hot take: I've seen hundred of projects without funding better maintained than this one.~

This fixed the issue for me.

**Comment by Blackilykat** [comment]
> My guess is that the issue happen when you're deleting some comments, so don't until futo take care of this issue..

Oh yeah, I hadn't made the connection but that checks out. this happened both times I tried deleting a comment.
I intended to create an issue about the comments not getting deleted but never came around to it.

**Comment by Generalaccountuse** [comment]
@[johnsmith744]
I totally agree with that.
That's why I was hoping Odysee would come out victor in the video platform war, because they have a block and mute button for every user.

Also, I know who you are talking about since they got me too.

**Comment by johnsmith744** [comment]
> > I finally found a fix tor this issue!
> > ```
> > 1. Settings > Scroll down > Backup > Export data manually
> > 
> > 2. Manage polycentric identity > Export > Copy backup string and store it in password manager
> > 
> > 3. Wipe all the app data in your phone's settings
> > 
> > 4. Open GrayJay
> > 
> > 5. Enable the Sources you were using before wiping data (or else they will all be enabled)
> > 
> > 6. Settings > Import Data > Choose the zip you have exported in step 1
> > 
> > 7. Follow instruction, Click No when asked to enable all sources (I find it annoying to have them all enabled).
> > 
> > 8. Settings > Manage polycentric id > Import your ID back into GrayJay by pasting the backup string
> > 
> > 9. YOU CAN FINALLY USE THIS APP AS INTENDED BACK AGAIN!
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > My guess is that the issue happen when you're deleting some comments, so don't until futo take care of this issue..
> > ~Hot take: I've seen hundred of projects without funding better maintained than this one.~
> 
> This fixed the issue for me.

These steps seem to resolve the issue.  It's possible del comment might cause the problem.  I've done some del because there is no option to "edit" to fix some stuff like spelling.

**Comment by cosmicrabbit452** [comment]
> I finally found a fix tor this issue!
Thanks for being the guinea pig, I wasn't going to risk losing my whole setup just to shootpost on gamer videos

**Comment by Assblastois** [comment]
I just made a new profile. I tried following the steps but exporting the polycentric ID was so buggy and for some reason the copied a string was so large that when pasting it, it would freeze my whole phone for at least a solid minute. Fuck this, just make a new ID and continue shitposting and sneeding.

**Comment by mathix420** [comment]
Yes, the export method for polycentric IDs is so shitty. Definitely not a good way of doing it, but since this is not really an open source project we can't even contribute..

**Comment by Blackilykat** [comment]
> This is just for issue tracking, development happens at gitlab.futo.org and the source is fully available

Source available isn't the same as open source though. You can't have an account on gitlab.futo.org without being an employee, so us users have no way of contributing.

**Comment by tom-futo** [comment]
> > This is just for issue tracking, development happens at gitlab.futo.org and the source is fully available
> 
> Source available isn't the same as open source though. You can't have an account on gitlab.futo.org without being an employee, so us users have no way of contributing.

We read and review PR's created in GitHub if you're interested in making a contribution.

**Comment by eddbird** [comment]
> It happened again for me, I think I have more insight to possibly find the source of the issue.
> 
>     1. Post a comment on a video
> 
>     2. Delete this comment
> 
>     3. Post again on the same video
> 
> 
> And suddenly none of your comments are being published...

I was about to post an entire new bug report before double checking. However, you don't have to post again on the same video. I encountered and reproduced this issue trying to delete any comment on any video, and then trying to post any comment on any subsequent video. Every time I clear the profile and then attempt to delete a comment, this bug occurs. It's Probably the cause of issue #1412 as well.

Near as I can tell this bug has been around since I started using grayjay with polycentric. This needs a fix ASAP.
